src/lib/mina_lib/mina_lib.ml

```diff
@@ -1948,7 +1948,7 @@ let create ~commit_id ?wallets (config : Config.t) =
               ~frontier_broadcast_writer:frontier_broadcast_pipe_w ~catchup_mode
               ~network_transition_reader:block_reader
               ~producer_transition_reader ~get_most_recent_valid_block
-              ~most_recent_valid_block_writer
+              ~most_recent_valid_block_writer ~snark_pool
```

Comment:
my intuition nudges me into thinking this should be a reference, in the case of the transition frontier we read from a pipe. i will figure out of this truly works while testing, but i am somewhat concerned.

Reply:
this worked dynamically during testing